File: apps/android/app/src/main/java/com/shadowchat/app/ShadowRepositoryProvider.kt
package com.shadowchat.app

import com.shadowchat.features.chatlist.ChatListRepository
import com.shadowchat.features.timeline.RoomTimelineRepository

internal interface ShadowRepositoryProvider {
    val chatListRepository: ChatListRepository

    fun roomTimelineRepository(roomId: String): RoomTimelineRepository
}

internal object DemoShadowRepositoryProvider : ShadowRepositoryProvider {
    override val chatListRepository: ChatListRepository = DemoChatListRepository

    override fun roomTimelineRepository(roomId: String): RoomTimelineRepository = DemoRoomTimelineRepository
}

File: apps/ios/Packages/Sources/ShadowChatAppShell/ShadowRepositoryProvider.swift
import ShadowChatListFeature
import ShadowRoomTimelineFeature

public protocol ShadowRepositoryProvider: Sendable {
    func makeChatListRepository() -> ChatListRepository
    func makeRoomTimelineRepository(roomId: String) -> RoomTimelineRepository
}

struct DemoShadowRepositoryProvider: ShadowRepositoryProvider {
    func makeChatListRepository() -> ChatListRepository {
        DemoChatListRepository()
    }

    func makeRoomTimelineRepository(roomId: String) -> RoomTimelineRepository {
        DemoRoomTimelineRepository()
    }
}
